astropy/io/fits/hdu/hdulist.py

```diff
@@ -512,6 +510,15 @@ def fileinfo(self, index):
 
         return output
 
+    def _load_all_hdus(self):
```

Comment:
Maybe `_read_all_hdus` to be consistent with `_read_next_hdu` ?

Reply:
Actually I just saw that there is a `HDUList.readall` method, implemented differently but the result should be the same.

Reply:
I just tested this, and it does appear to work, except when substituting `readall` for `_read_all_hdus` in the `__len__` function. Should I go ahead and change it to `readall` and keep the `while` loop in `__len__`, or leave it as is?

Reply:
I think `readall` could use the `while self._read_next_hdu(): pass` method as it is more efficient (no need to truly iterate and go to `__getitem__` etc. since the goal is just to force the loading). I guess Erik forgot about this `readall` when adding the lazy loading (last modification of  `readall` was in 2011). This was we avoid to have two "readall" methods, and we can use it as you did with your new method.